selftests/unit/test_utils_vmimage.py

```diff
@@ -13,5 +19,101 @@ def test_concrete_providers_have_name(self):
             self.assertTrue(hasattr(provider, 'name'))
 
 
+class VMImageHtmlParserTest(unittest.TestCase):
+    def setUp(self):
+        pattern = '^[0-9]+/$'
+        self.parser = vmimage.VMImageHtmlParser(pattern)
+
+    def test_handle_starttag_no_a(self):
+        self.parser.feed('<html><head><title>Test</title></head>'
+                         '<body><h1>VMImage</h1></body></html>')
+        self.assertFalse(self.parser.items)
+
+    def test_handle_starttag_with_a_pattern_not_match(self):
+        self.parser.feed('<html><head><title>Test</title></head>'
+                         '<body><a href="https://test.com/" /></body></html>')
+        self.assertFalse(self.parser.items)
+
+    def test_handle_starttag_with_a_pattern_match(self):
+        version = '12'
+        self.parser.feed('<html><head><title>Test</title></head>'
+                         '<body><a href="%s/" /></body></html>' % version)
+        self.assertTrue(self.parser.items)
+        self.assertEqual(self.parser.items[0], version)
+
+    def test_handle_starttag_with_a_pattern_match_multiple(self):
+        versions = ['12', '13', '14']
+        links = ['<a href="%s/" />' % version for version in versions]
+        html = ('<html><head><title>Test</title></head>'
+                '<body>%s</body></html>' % ''.join(links))
+        self.parser.feed(html)
+        self.assertTrue(self.parser.items)
+        self.assertEqual(len(self.parser.items), len(versions))
+        self.assertTrue(all(version in self.parser.items for version in versions))
+
+
+class ImageProviderBaseTest(unittest.TestCase):
+    def _get_html_with_versions(self, versions):
```

Comment:
This can be a staticmethod.

Reply:
Sure, I can change it. What's the advantage to have this method as static? I know it doesn't use the instance (self) for anything, but beyond that is there any real advantage having static method (when it could be) over a common method?